src/Model/BaseCollection.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace OpenEuropa\CdtClient\Model;
+
+/**
+ * Class BaseCollection.
+ *
+ * A base class for collections.
+ *
+ * @extends \ArrayIterator<int|string, mixed>
+ */
+abstract class BaseCollection extends \ArrayIterator
```

Comment:
I created this class as a base of all class collection, it allows strict typing of all methods. Please let me know if there is a smoother way to create an iterator.